File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.py
def divisor(n, limit, power):
    cnt = 0 
    for i in range(1, int(n**0.5)+1):
        if i*i == n:
            cnt += 1
        elif n%i == 0:
            cnt += 2
        
        if cnt > limit:
            return power
    return cnt
    
def solution(number, limit, power):
    result = 0
    for i in range(1, number+1):
        d = divisor(i, limit, power)
        result += d
        
    return result
        

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.java
class Solution {
    public int solution(int number, int limit, int power) {
        int totalWeight = 0; 

        for (int i = 1; i <= number; i++) {
            int attackPower = countDivisors(i);

            if (attackPower > limit) {
                totalWeight += power;
            } else {
                totalWeight += attackPower;
            }
        }

        return totalWeight; 
    }

    public int countDivisors(int n) {
        int count = 0;

        for (int i = 1; i <= Math.sqrt(n); i++) {
            if (n % i == 0) {
                if (i * i == n) {
                    count += 1; 
                } else {
                    count += 2;
                }
            }
        }

        return count;
    }
}
